client/packages/system/src/MasterList/Components/MasterListSearchModal/MasterListSearchModal.tsx

```diff
@@ -24,12 +24,11 @@ export const MasterListSearchModal: FC<MasterListSearchProps> = ({
     sortBy,
     filterBy
   );
-  const t = useTranslation(['app', 'common']);
+  const t = useTranslation('app');
 
-  // Only query for data once the modal has been opened
   useEffect(() => {
-    if (open && !data && !isLoading) mutate();
-  }, [open, data, isLoading]);
+    if (open) mutate();
```

Comment:
Data is cached, so previous check will always return false after first fetch... couldn't get it to clear in the hook, so have just changed the condition to re-fetch every time the modal opens...

Reply:
We should really have a query version of listAll. In `useQuery` there is the option to enable or disable the query, i.e. you can pass `enabled: open` as a parameter. react/tan query should do the rest for you. @mark-prins or am I missing something here?

Example to enable the query: `packages/common/src/authentication/api/hooks/useUserDetails.ts`

Reply:
The useMasterListAll is using a mutation to get the query... I mean, I could change it to useQuery if there wasn't a reason for it to be useMutation

Reply:
yeah, thanks @clemens-msupply, was wondering about doing that. 

The usage in `CreateStocktakeButton` is using `useEffect` to do what a `useQuery` would give you anyway - ie that would be better as a `useQuery`.

I was reluctant to change to a query because, like you say Roxy, all the `useAll` functions are using mutations. and I'm also reluctant to duplicate the methods ( but I'll get over that ). Can't just use `useMasterList.document.list` as that is using url query params. 

I think there's no perfect option - what you've done with the `open` var is my top pick which is why I approved. If you want to change, my preference is the option of changing `listAll` to use `useQuery` and then updating the three usages. But any of the options would be fine.